lib/elixir/lib/code.ex

```diff
@@ -2028,7 +2028,11 @@ defmodule Code do
   @spec fetch_docs(module | String.t()) ::
           {:docs_v1, annotation, beam_language, format, module_doc :: doc_content, metadata,
            docs :: [doc_element]}
-          | {:error, :module_not_found | :chunk_not_found | {:invalid_chunk, binary}}
+          | {:error,
+             :module_not_found
+             | :chunk_not_found
+             | {:invalid_chunk, binary}
+             | :beam_lib.chnk_rsn()}
```

Comment:
Not all errors are possible here and we already handle some of those, so I would prefer to explicitly list any other case we may have seen. What is the additional error you ran into?

Reply:
> What is the additional error you ran into?

Quick search in my notes indicates `{:error, :beam_lib, {:not_a_beam_file, ""}}`

> Not all errors are possible here and we already handle some of those, so I would prefer to explicitly list any other case we may have seen

It's a filesystem read, it can fail for a number of reasons on posix and windows. The user may e.g. eject the drive

Reply:
Sure, I am fine with adding something like:

    {:error, {:file_error, :file.filename(), :file.posix()}}

But I don't think we should add all of the beam_lib exceptions as is.

We can also add `:invalid_beam` to deal with "not_a_beam_file" and "invalid_beam_file". :)

Reply:
I'm OK with catch-all `:invalid_beam`

Reply:
@lukaszsamson sounds good to me!